implementation/src/main/java/io/smallrye/health/InetAddressHealthCheck.java

```diff
@@ -0,0 +1,52 @@
+package io.smallrye.health;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Health check implementation to check if host is reachable using {@code java.net.InetAddress.isReachable} method.
+ *
+ * <pre>
+ * {@code
+ * @Produces
+ * @ApplicationScoped
+ * @Liveness
+ * HealthCheck check1() {
+ *   return new InetAddressHealthCheck("service.com");
+ * }
+ * }
+ * </pre>
+ *
+ * @see InetAddress
+ */
+public class InetAddressHealthCheck implements HealthCheck {
+
+    private String host;
+    private int timeout = 2000;
+
+    public InetAddressHealthCheck(String host) {
+        this.host = host;
+    }
+
+    public InetAddressHealthCheck(String host, int timeout) {
+        this.host = host;
+        this.timeout = timeout;
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        final HealthCheckResponseBuilder healthCheckResponseBuilder = HealthCheckResponse.named("Address %s reachable");
```

Comment:
Is the `%s` interpolated somewhere?

Reply:
I missed the String.format but well this PR is just to see if it is good idea or not, then I can polish this and add tests, ...